esphome/components/mitsubishi_itp/mitsubishi_uart-packetprocessing.cpp

```diff
@@ -355,12 +381,87 @@ void MitsubishiUART::process_packet(const RemoteTemperatureSetRequestPacket &pac
 
     publish_on_update_ |= (old_thermostat_temp != thermostat_temperature_sensor_->raw_state);
   }
-};
+}
+
+void MitsubishiUART::process_packet(const KumoThermostatSensorStatusPacket &packet) {
+  if (!kumo_emulation_mode_) return;
+
+  ESP_LOGV(TAG, "Processing inbound %s", packet.to_string().c_str());
+
+  if (thermostat_humidity_sensor_ && packet.get_flags() & 0x04) {
+    const float old_humidity = thermostat_humidity_sensor_->raw_state;
+    thermostat_humidity_sensor_->raw_state = packet.get_indoor_humidity_percent();
+    publish_on_update_ |= (old_humidity != thermostat_humidity_sensor_->raw_state);
+  }
+
+  if (thermostat_battery_sensor_ && packet.get_flags() & 0x08) {
+    const auto old_battery = thermostat_battery_sensor_->raw_state;
+    thermostat_battery_sensor_->raw_state = THERMOSTAT_BATTERY_STATE_NAMES[packet.get_thermostat_battery_state()];
+    publish_on_update_ |= (old_battery != thermostat_battery_sensor_->raw_state);
+  }
+
+  ts_bridge_->send_packet(SetResponsePacket());
+}
+
+void MitsubishiUART::process_packet(const KumoThermostatHelloPacket &packet) {
+  if (!kumo_emulation_mode_) return;
+
+  ESP_LOGV(TAG, "Processing inbound %s", packet.to_string().c_str());
+
+  ts_bridge_->send_packet(SetResponsePacket());
+}
+
+void MitsubishiUART::process_packet(const KumoThermostatStateSyncPacket &packet) {
+  if (!kumo_emulation_mode_) return;
+
+  ESP_LOGV(TAG, "Processing inbound %s", packet.to_string().c_str());
+
+  if (packet.get_flags() & 0x08) this->last_heat_setpoint_ = packet.get_heat_setpoint();
+  if (packet.get_flags() & 0x10) this->last_cool_setpoint_ = packet.get_cool_setpoint();
+
+  ts_bridge_->send_packet(SetResponsePacket());
+}
+
+void MitsubishiUART::process_packet(const KumoAASetRequestPacket &packet) {
+  if (!kumo_emulation_mode_) return;
+
+  ESP_LOGV(TAG, "Processing inbound KumoAASetRequestPacket: %s", packet.to_string().c_str());
+
+  ts_bridge_->send_packet(SetResponsePacket());
+}
+
 void MitsubishiUART::process_packet(const SetResponsePacket &packet) {
   ESP_LOGV(TAG, "Got Set Response packet, success = %s (code = %x)", packet.is_successful() ? "true" : "false",
            packet.get_result_code());
   route_packet_(packet);
 }
 
+// Process incoming data requests (Kumo)
+void MitsubishiUART::handle_kumo_adapter_state_get_request(const GetRequestPacket &packet) {
+  if (!kumo_emulation_mode_) return;
+
+  auto response = KumoCloudStateSyncPacket();
+
+  response.set_heat_setpoint(this->last_heat_setpoint_);
+  response.set_cool_setpoint(this->last_cool_setpoint_);
```

Comment:
What happens here if these are NAN?

Reply:
Good point. I've changed the code to send 0x00 in this case, which doesn't seem to cause the MHK to complain any.